cli/commands/scaffold/action.go

```diff
@@ -0,0 +1,392 @@
+package scaffold
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/shell"
+
+	"github.com/gruntwork-io/terragrunt/terraform"
+
+	"github.com/gruntwork-io/terragrunt/cli/commands/hclfmt"
+	"github.com/gruntwork-io/terragrunt/util"
+
+	boilerplate_options "github.com/gruntwork-io/boilerplate/options"
+	"github.com/gruntwork-io/boilerplate/templates"
+	"github.com/gruntwork-io/boilerplate/variables"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terratest/modules/files"
+	"github.com/hashicorp/go-getter"
+)
+
+const (
+	moduleCount            = 2
+	moduleAndTemplateCount = 3
+
+	sourceUrlTypeHttps = "git-https"
+	sourceUrlTypeGit   = "git-ssh"
+	sourceGitSshUser   = "git"
+
+	sourceUrlTypeVar    = "SourceUrlType"
+	sourceGitSshUserVar = "SourceGitSshUser"
+	refVar              = "Ref"
+	// refParam - ?ref param from url
+	refParam = "ref"
+
+	moduleUrlPattern = `git::([^:]+)://([^/]+)(/.*)`
+	moduleUrlParts   = 4
+
+	defaultBoilerplateConfig = `
+variables:
+
+`
+	defaultTerragruntTemplate = `
```

Comment:
I forgot one thing to include in this default template:

```hcl
include "root" {
  path = find_in_parent_folders()
}
```

Could you add that just below the `terraform { }` block? And please wrap it in an if statement, where it only renders if the `EnableRootInclude` variable is true. This variable should default to `true`, but users can shut it off if they don't want it via `-var EnableRootInclude=false`.